go/fn/subobject.go

```diff
@@ -35,3 +36,55 @@ func (o *SubObject) GetMap(k string) *SubObject {
 	}
 	return &SubObject{obj: m}
 }
+
+// GetSlice gets the component in the `fields` path. It expects the component to be
+// a slice of SubObjects.
+func (o *SubObject) GetSlice(fields ...string) (SliceSubObjects, bool, error) {
+	var mapVariant *internal.MapVariant
+	if len(fields) > 1 {
+		m, found, err := o.obj.GetNestedMap(fields[:len(fields)-1]...)
+		if err != nil || !found {
+			return nil, found, err
+		}
+		err = m.Node().Decode(mapVariant)
+		return nil, true, err
+	} else {
+		mapVariant = o.obj
+	}
+	sliceVal, found, err := mapVariant.GetNestedSlice(fields[len(fields)-1])
+	if err != nil {
+		panic(ErrSubObjectFields{fields: fields})
+	}
+	if !found {
+		return nil, found, nil
+	}
+	objects, err := sliceVal.Objects()
```

Comment:
It seems `Objects()` returns a list of `*internal.MapVariant` which is neither `KubeObject` nor `SubObject`. Maybe we can consider changing its name or make it private.

Reply:
Yes, the sliceVal and mapVariant are all internal so users won't call Objects directly. but that's a good point we shall rename Objects to something else, How about `Elements()`

Reply:
Changed both sliceVariant and doc